fix ratio modifier application

jnq-t opened the pull request:
we were applying the ratio to the input frequency of the scale degree rather than to the starting frequency of the whole scale. applied a fix. @pluto-bell  i can't figure out how to test now that you added the FE stuff 🤣  maybe we could test this tgth before i merge it?

Commit:
ratio modifier fix

pluto-bell commented on app/TEMP_classes_workspace.js:
```diff
@@ -66,17 +66,12 @@ class ScaleDegree {
-        frequency *= Math.pow(2, this.modifier.detuneByCents/1200); // if no detune is provided this will be 1
-        return frequency;
+        if (this.modifier.ratioNumerator > 0 && this.modifier.ratioDenominator > 0) return this.#ratio(this.startingFreq);
+        return this.inputFrequency *= Math.pow(2, this.modifier.detuneByCents/1200); // if no detune is provided this will be 1
```
make helper method for detune by cents

jnq-t replied:
got it!

pluto-bell commented on app/TEMP_classes_workspace.js:
```diff
@@ -66,17 +66,12 @@ class ScaleDegree {
-        frequency *= ratio; // if no ratio is provided this will be 1
-        frequency *= Math.pow(2, this.modifier.detuneByCents/1200); // if no detune is provided this will be 1
-        return frequency;
+        if (this.modifier.ratioNumerator > 0 && this.modifier.ratioDenominator > 0) return this.#ratio(this.startingFreq);
```
re-write so both ratio & detune by cents can be used

jnq-t replied:
adding this to the backlog

pluto-bell reviewed (changes requested):
Please review my comments (:

Commit:
feedback

pluto-bell commented:
### All freq.'s getting altered

**Documents on**
`(78) [242.7886656724614, 257.22563100871037, 272.5210629770106, 288.72600865970804, 305.89455055662296, 300, 343.3550237844621, 363.7719759624957, 385.40298329442084, 420, 432.6002220531987, 458.32396987700014, 485.5773313449228, 514.4512620174207, 545.0421259540212, 577.4520173194161, 611.7891011132459, 600, 686.7100475689242, 727.5439519249913...]`


**Documents off**
`(77) [240, 254.27114264623089, 269.3908915942495, 285.40970760065306, 302.3810519747696, 320.3615650008083, 339.4112549695429, 359.5936984504036, 380.97625247236795, 403.63027932178306, 427.631384707363, 453.059670087213, 480, 508.54228529246177, 538.781783188499, 570.8194152013061, 604.7621039495392, 640.7231300016166, 678.8225099390858...]`

Commit:
debug ratio

was multiplying the input frequency of the scale degree by the ratio, rather than the starting frequency of the whole scale by the ratio. this was making ridiculously high frequencies and we were hitting the cuttoff
